analysis/filter/MergeFilter/src/main/kotlin/de/maibornwolff/codecharta/filter/mergefilter/RecursiveNodeMergerStrategy.kt

```diff
@@ -54,8 +54,8 @@ class RecursiveNodeMergerStrategy(ignoreCase: Boolean = false): NodeMergerStrate
     }
 
     private fun merge(vararg nodes: MutableNode): MutableNode {
-        val node = nodes[0].merge(nodes.toList())
-        node.children.addAll(this.mergeNodeLists(nodes.map { it.children }))
+        val node = nodes.toList()[0].merge(nodes.toList())
+        node.children.addAll(this.mergeNodeLists(nodes.map { it.children.toList() }.toList()))
```

Comment:
Do we have to first convert the nodes to a list? Or might there be another possibility to do this directly?